### PR TITLE
drivers: uart: npcx: use PM constraint for the receiving

### PR DESCRIPTION
**Pull request by ChiHuaL**
The driver sets the PM constraint to prevent the system
from entering the suspend state for a CONFIG_UART_CONSOLE_INPUT_EXPIRED
period when data come in. The constraint releases after the
CONFIG_UART_CONSOLE_INPUT_EXPIRED time expires, .

With this PR, the PM policy doesn't have to check the timestamp
by calling npcx_power_console_is_in_use() explictly. So the related
npcx_power_console_is_in_use*() functions can be removed.

This PR also fixes the coding style of the driver.

Signed-off-by: Jun Lin <CHLin56@nuvoton.com>
Signed-off-by: Wealian Liao <WHLIAO@nuvoton.com>

### CODE REVIEW

File: drivers/serial/uart_npcx.c [hunk 1]
```diff
@@ -57,20 +69,22 @@ struct uart_npcx_data {
 	(struct uart_reg *)(DRV_CONFIG(dev)->uconf.base)
 
 #if defined(CONFIG_PM) && defined(CONFIG_UART_INTERRUPT_DRIVEN)
-static void uart_npcx_pm_constraint_set(const struct device *dev)
+static void uart_npcx_pm_constraint_set(const struct device *dev,
```
Comment (keith-zephyr) [inline drivers/serial/uart_npcx.c:72]: If you change the parameter of the constraint helper routines to `struct uart_npcx_data *data`, then you can eliminate the `data->uart_dev` field.

File: drivers/serial/uart_npcx.c [hunk 2]
```diff
@@ -365,6 +393,17 @@ static __unused void uart_npcx_rx_wk_isr(const struct device *dev,
 	npcx_uart_disable_access_interrupt();
 }
 
+#ifdef CONFIG_UART_CONSOLE_INPUT_EXPIRED
+static void uart_npcx_rx_refresh_timeout(struct k_work *work)
+{
+	struct uart_npcx_data *data = CONTAINER_OF(work, struct uart_npcx_data,
+						   rx_refresh_timeout_work);
+	const struct device *dev = data->uart_dev;
```
Comment (keith-zephyr) [inline drivers/serial/uart_npcx.c:401]: See above, change `uart_npcx_pm_constraint_rel/set` to accept `struct uart_npcx_data` and you can eliminate this.

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

File: drivers/serial/uart_npcx.c [hunk 3]
```diff
@@ -42,7 +49,12 @@ struct uart_npcx_data {
 	void *user_data;
 #endif
 #ifdef CONFIG_PM
-	atomic_t pm_constraint_on;
+	ATOMIC_DEFINE(pm_constraint_flag, UART_PM_CONSTRAINT_FLAG_COUNT);
+#ifdef CONFIG_UART_CONSOLE_INPUT_EXPIRED
+	atomic_t pm_rx_constraint_on;
```
[the remark this answers is not in the document]
Comment (ChiHuaL) [reply drivers/serial/uart_npcx.c]: Done.

File: drivers/serial/uart_npcx.c (hunk 1 above)
Comment (ChiHuaL) [reply drivers/serial/uart_npcx.c]: Done.

File: drivers/serial/uart_npcx.c (hunk 2 above)
Comment (ChiHuaL) [reply drivers/serial/uart_npcx.c]: Done.

File: drivers/serial/uart_npcx.c [hunk 4]
```diff
@@ -28,10 +28,17 @@ struct uart_npcx_config {
 	/* int-mux configuration */
 	const struct npcx_wui uart_rx_wui;
 	/* pinmux configuration */
-	const uint8_t   alts_size;
```
Comment (jackrosenthal) [inline drivers/serial/uart_npcx.c:31]: Thanks for fixing this. Can you move the unrelated style changes to a separate commit?

File: drivers/serial/uart_npcx.c [hunk 5]
```diff
@@ -81,7 +89,7 @@ static int uart_set_npcx_baud_rate(struct uart_reg *const inst, int baud_rate,
 				   int src_clk)
 {
 	/* Fix baud rate to 115200 so far */
-	if (baud_rate  == 115200) {
```
Comment (jackrosenthal) [inline drivers/serial/uart_npcx.c:84]: Same here

**Review by jackrosenthal — APPROVED** [review]

File: drivers/serial/uart_npcx.c (hunk 4 above)
Comment (ChiHuaL) [reply drivers/serial/uart_npcx.c:31]: Ok. I have separated the coding style to the second commit.

File: drivers/serial/uart_npcx.c (hunk 5 above)
Comment (ChiHuaL) [reply drivers/serial/uart_npcx.c:84]: done.

**Review by MulinChao — APPROVED** [review]

**Comment by ChiHuaL** [comment]
@keith-zephyr Do you have suggestion on this PR? Any comment is appreciated.

**Review by keith-zephyr — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Comment by ChiHuaL** [comment]
Could we  get the PR merged?
